In [27]:
import pandas as pd
import numpy as np


Reading the data:

In [28]:
df = pd.read_csv("E:\Guvi\Projects\Inventory forecasting\\train 2.csv")
df

date  store  item  sales
0       2013-01-01      1     1     13
1       2013-01-02      1     1     11
2       2013-01-03      1     1     14
3       2013-01-04      1     1     13
4       2013-01-05      1     1     10
...            ...    ...   ...    ...
912995  2017-12-27     10    50     63
912996  2017-12-28     10    50     59
912997  2017-12-29     10    50     74
912998  2017-12-30     10    50     62
912999  2017-12-31     10    50     82

[913000 rows x 4 columns]

Exploratory Data Analysis(EDA)

In [29]:
df.nunique()

date     1826
store      10
item       50
sales     213
dtype: int64

findings: 
We can see that the data is clean of outliers and duplicates. Since, there will be 
1. 1826 days in 5 years period 
2. As per the data we have 10 unique stores and 50 unique products. 
3. We can not tell whether all the values in sales are right/wrong.

In [30]:
df.dtypes

date     object
store     int64
item      int64
sales     int64
dtype: object

Finding: We can see that all the columns are in 'int64' datatype other than date which is in 'object' datatype.

In [31]:
df['date'] = pd.to_datetime(df['date'])
df.dtypes

date     datetime64[ns]
store             int64
item              int64
sales             int64
dtype: object

Changed date column to 'datetime' format for better interpretation. 

In [32]:
print(df)

             date  store  item  sales
0      2013-01-01      1     1     13
1      2013-01-02      1     1     11
2      2013-01-03      1     1     14
3      2013-01-04      1     1     13
4      2013-01-05      1     1     10
...           ...    ...   ...    ...
912995 2017-12-27     10    50     63
912996 2017-12-28     10    50     59
912997 2017-12-29     10    50     74
912998 2017-12-30     10    50     62
912999 2017-12-31     10    50     82

[913000 rows x 4 columns]


In [33]:
x = df.groupby(['date','store','item']).sum()
x

sales
date       store item       
2013-01-01 1     1        13
                 2        33
                 3        15
                 4        10
                 5        11
...                      ...
2017-12-31 10    46       70
                 47       22
                 48       60
                 49       42
                 50       82

[913000 rows x 1 columns]

In [34]:
df2 = df.groupby(['date','item'], as_index = False).sales.sum()
df2

date  item  sales
0     2013-01-01     1    133
1     2013-01-01     2    327
2     2013-01-01     3    172
3     2013-01-01     4    102
4     2013-01-01     5     83
...          ...   ...    ...
91295 2017-12-31    46    605
91296 2017-12-31    47    211
91297 2017-12-31    48    494
91298 2017-12-31    49    285
91299 2017-12-31    50    626

[91300 rows x 3 columns]

In [35]:
df3 = df2.groupby(['item','date'],as_index = False).sales.sum()

In [36]:
df3

item       date  sales
0         1 2013-01-01    133
1         1 2013-01-02     99
2         1 2013-01-03    127
3         1 2013-01-04    145
4         1 2013-01-05    149
...     ...        ...    ...
91295    50 2017-12-27    511
91296    50 2017-12-28    587
91297    50 2017-12-29    596
91298    50 2017-12-30    612
91299    50 2017-12-31    626

[91300 rows x 3 columns]

In [37]:
# df_item_1 = df3[df3['item']==1]
# df_item_1 

In [52]:
df_items = []
for i in range(0,51):
    df_item_i = df3[df3['item']==i]
    df_items.append(df_item_i) 

In [38]:
# df3['demand'] = df2.sales.rolling(window = 90).sum()

In [39]:
# df3

In [40]:
# df3.iloc[89:]

In [49]:
empty_var = ['df_item_{}'.format(i) for i in range(1,51)]

In [50]:
print(empty_var)

['df_item_1', 'df_item_2', 'df_item_3', 'df_item_4', 'df_item_5', 'df_item_6', 'df_item_7', 'df_item_8', 'df_item_9', 'df_item_10', 'df_item_11', 'df_item_12', 'df_item_13', 'df_item_14', 'df_item_15', 'df_item_16', 'df_item_17', 'df_item_18', 'df_item_19', 'df_item_20', 'df_item_21', 'df_item_22', 'df_item_23', 'df_item_24', 'df_item_25', 'df_item_26', 'df_item_27', 'df_item_28', 'df_item_29', 'df_item_30', 'df_item_31', 'df_item_32', 'df_item_33', 'df_item_34', 'df_item_35', 'df_item_36', 'df_item_37', 'df_item_38', 'df_item_39', 'df_item_40', 'df_item_41', 'df_item_42', 'df_item_43', 'df_item_44', 'df_item_45', 'df_item_46', 'df_item_47', 'df_item_48', 'df_item_49', 'df_item_50']


In [54]:
df_items

[Empty DataFrame
 Columns: [item, date, sales]
 Index: [],
       item       date  sales
 0        1 2013-01-01    133
 1        1 2013-01-02     99
 2        1 2013-01-03    127
 3        1 2013-01-04    145
 4        1 2013-01-05    149
 ...    ...        ...    ...
 1821     1 2017-12-27    154
 1822     1 2017-12-28    178
 1823     1 2017-12-29    207
 1824     1 2017-12-30    236
 1825     1 2017-12-31    227
 
 [1826 rows x 3 columns],
       item       date  sales
 1826     2 2013-01-01    327
 1827     2 2013-01-02    279
 1828     2 2013-01-03    304
 1829     2 2013-01-04    359
 1830     2 2013-01-05    377
 ...    ...        ...    ...
 3647     2 2017-12-27    500
 3648     2 2017-12-28    495
 3649     2 2017-12-29    515
 3650     2 2017-12-30    538
 3651     2 2017-12-31    634
 
 [1826 rows x 3 columns],
       item       date  sales
 3652     3 2013-01-01    172
 3653     3 2013-01-02    213
 3654     3 2013-01-03    193
 3655     3 2013-01-04    218
 3656     3 201

In [55]:
map(df_items,empty_var)

In [58]:
empty_var[1]

'df_item_2'